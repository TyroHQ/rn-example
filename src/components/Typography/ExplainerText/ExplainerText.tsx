import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";

const ExplainerTextContainer = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class ExplainerText extends React.Component<{}> {
  static defaultProps: ExplainerText["props"];
  render() {
    return <ExplainerTextContainer>ExplainerText</ExplainerTextContainer>;
  }
}
