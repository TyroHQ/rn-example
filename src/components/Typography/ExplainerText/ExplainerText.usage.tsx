import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { ExplainerText } from "./ExplainerText";

export const basic = () => <ExplainerText />;
