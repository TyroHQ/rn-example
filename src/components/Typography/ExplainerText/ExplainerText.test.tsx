import { shallow } from "enzyme";
import * as usage from "./ExplainerText.usage";

describe("<ExplainerText />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
