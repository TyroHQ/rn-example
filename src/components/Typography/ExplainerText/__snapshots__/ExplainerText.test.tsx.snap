// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExplainerText /> renders well in basic mode 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  ExplainerText
</Text>
`;
